lib/services/useracl.go

```diff
@@ -153,6 +152,14 @@ func NewUserACL(user types.User, userRoles RoleSet, features proto.Features, des
 	cnDiagnosticAccess := newAccess(userRoles, ctx, types.KindConnectionDiagnostic)
 	samlIdpServiceProviderAccess := newAccess(userRoles, ctx, types.KindSAMLIdPServiceProvider)
 
+	// active sessions are a special case - if a user's role set has any join_sessions
+	// policies then the ACL must permit showing active sessions
```

Comment:
just to be clear,
with access to other resources, a deny role "outweighs" any allow roles and will deny access to that resource if it exists. but with the case of active sessions, it's the opposite?

Reply:
Right, this is a bit of a special case. The RBAC resource for "KindSSHSession" was the old way of granting access to join sessions, and has been deprecated by the `join_sessions` policy in roles.

We made this change to RBAC in Teleport 9 or 10, but never updated the web UI's ACL rule.

As a result, the web UI would sometimes incorrectly hide the "Active Sessions" page even if you did have permission to join them. You can confirm this behavior by creating a deny '*' rule like the one in the test in this PR, and manually crafting the URL to join the session - you'll see that Teleport RBAC lets you in just fine, and it was a web UI bug that was preventing the page from showing up.